test_suite.py

```diff
@@ -103,6 +103,23 @@ def test_torch_summer():
         < 0.0001
     ), "The summer is not working properly!"
 
+def test_torch_2D_fixed_speedup(sample_size=10_000):
+    device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+    data = torch.randn(32,8,8,64).to(device)
+
+    # Fixed 
+    pos_enc = FixedPositionalEncoding2D((8,32), 64).to(device)
+    start_time = time.time()
+    for i in range(sample_size):
+        pos_enc(data)
+    print(f"Duration when fixed: {time.time()- start_time}")
+
+    # Original
+    pos_enc = PositionalEncoding2D(64).to(device)
+    start_time = time.time()
+    for i in range(sample_size):
+        pos_enc(data)
+    print(f"Duration of original: {time.time()- start_time}")
```

Comment:
I am not so sure that the print does anything when ran with `pytest -vv`, but i could be wrong.

In any case, it would be cool to have an assert here on something. I am not sure on what exactly, since i don't think asserting on time is a good idea (since the test will probably be flaky), but still it would be cool to have something here.

Reply:
yeah, this was just a place holder in order to let you test the speed up. But I have the idea to confirm that the output of the fixed version matches the output of the original version. What do you think of that test?

Reply:
I see you did that already, nice job :+1:

Reply:
Thank you